src/vehicle.py

```diff
@@ -0,0 +1,150 @@
+        ]
+        self.weapon_index = 0
+        self.primary_weapon: Weapon
+        self.secondary_weapon: Weapon
+        self.primary_weapon_sprite: arcade.Sprite
+        self.secondary_weapon_sprite: arcade.Sprite
+        self._swap_in_weapons()
+        self.x_shift = float
+        self.y_shift = float
+
+        self.drive_mode_index = 0
+        self.drive_modes = create_drive_modes(self)
+        self.velocity = (0.0, 0.0)
+        "Translational velocity -- (x,y) tuple -- measured in pixels per second"
+
+        self.movement = MovementControls(
+            LinkedSprite[Vehicle](texture=RED_CAR, scale=0.18)
+        )
+
+    def update(self, delta_time: float):
+        #
+        # Driving and movement
+        #
+        if self.player.alive:
+            self.movement.drive_input(delta_time, self, self.input)
+        self.movement.move(delta_time, self, self.sprite_lists.walls)
+
+        #
+        # Weapons
+        #
+        if self.player.alive:
+            self.primary_weapon.update(delta_time)
+            self.secondary_weapon.update(delta_time)
+            if self.input.swap_weapons_button.pressed:
+                self._swap_weapons()
+
+    def apply_damage(self, damage: float):
+        self.player.take_damage(damage)
+
+    def _swap_weapons(self):
+        # Moves the current secondary weapon to the primary weapon slot and the next weapon on the list becomes the secondary weapon
+        self._swap_out_weapons()
+        self._swap_in_weapons()
+
+    def _swap_out_weapons(self):
+        self.primary_weapon.swap_out()
+        self.secondary_weapon.swap_out()
+
+    def _swap_in_weapons(self):
+        self.primary_weapon = self.weapons_list[self.weapon_index](
+            self.sprite_lists,
+            self.input.primary_fire_button,
+            self.primary_weapon_transform,
+        )
+        self.weapon_index += 1
+        if self.weapon_index >= len(self.weapons_list):
+            self.weapon_index = 0
+        self.secondary_weapon = self.weapons_list[self.weapon_index](
+            self.sprite_lists,
+            self.input.secondary_fire_button,
+            self.secondary_weapon_transform,
+        )
+
+    @property
+    def location(self):
+        return get_sprite_location(self.sprite)
+
+    @location.setter
+    def location(self, location: tuple[float, float, float]):
+        set_sprite_location(self.sprite, location)
+        move_sprite_relative_to_parent(
+            self.primary_weapon.weapon_sprite,
+            self.sprite,
+            self.primary_weapon_transform,
+        )
+        move_sprite_relative_to_parent(
+            self.secondary_weapon.weapon_sprite,
+            self.sprite,
+            self.secondary_weapon_transform,
+        )
+
+    # these properties are for the convenience of calling vehicle.center_x rather than vehicle.location[0]
+    # they specifically don't have setters because location should be set instead to move the weapons as well
```

Comment:
I don't always love needing to set position and rotation all at once with `.location = (x, y, radians)`  But one nice thing, is it's a bit more efficient with CPU since it only moves the sprites and weapons once.  It's premature optimization, but still nice.

Reply:
Fixed. I added the setters in so they're there if you need them. They will also update the weapon positions

Reply:
Oh I actually was fine with not having the setters.  I'll try to be clearer next time, my bad.

Reply:
Doesn't hurt to have them